packages/web-extension/src/walletManager/walletManager.types.ts

```diff
@@ -8,11 +11,15 @@ export interface WalletManagerProps {
 export interface WalletManagerActivateProps<P extends string | number = string, O = unknown> {
   /** User given name for the observableWallet being activated */
   observableWalletName: string;
-  /**
-   * Internal unique id calculated by {@link WalletManagerUi} based on the keyManager `chainId` and
-   * the root public key hash.
-   */
-  walletId: string;
+
+  /** The walletId of the wallet to activate */
+  walletId: WalletId;
+
+  accountIndex?: number;
+
+  /** The chainId of the network to activate the wallet in */
+  chainId: Cardano.ChainId;
```

Comment:
WalletManager already knows chainId. Is it supposed to switch network when you activate? Maybe it's better to keep activation just for wallet id, and switch network separately?
```suggestion
```

Reply:
So in order to activate a wallet we need to retrieve its state from the storage and build the providers, for this we need all three arguments, walletId, chainId and account index. This was how it worked before too btw, is just that the client code would build the store id (pubkeyhash + networkId). We are now passing the pieces separated and the wallet manager is now in charge of building the storeage ID.

I also think this makes sense, the same credentials is still a different wallet if used in a different network. If we just take the wallet ID here, we would need to guess which network id and account index the client means to activate, which we could guess by looking at the currently activa wallet and reuse the same network id and account index, but we could run into problems, lets say the user has two wallets, in one wallet he has 3 accounts and in the second wallet has 1 account, what should we do here if he is currently at account 3 on the first wallet and switches to the first wallet? we could see the accounts are different and default to account 0 (or always do this), but ultimately I think Lace should be explicit about what wallet he means to activate (which credentials, which account and in which network), and also Lace knows this, because the user will trigger this action via the GUI so Lace knows exactly what wallet the user wants (credentials, network, account). I added the utility method "switchNetwork" which does this guess work and reuses the current active wallet properties and just tries to activate with a different network.

But let me know what you think, if you dont agree with this approach I can add the logic to make it work like you suggest